### PR TITLE
Added the amount of resources

### PR DESCRIPTION
**Pull request by fatbasstard**
Issue https://github.com/awslabs/cfn-python-lint/issues/75

Add the amount of resources so it's clear how many resources there are

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/cfnlint/rules/resources/LimitNumber.py [hunk 1]
```diff
@@ -35,7 +35,7 @@ def match(self, cfn):
         # Check number of resources against the defined limit
         resources = cfn.get_resources()
         if len(resources) > LIMITS['resources']['number']:
-            message = 'Maximum number of resources ({0}) exceeded'
-            matches.append(RuleMatch(['Resources'], message.format(LIMITS['resources']['number'])))
+            message = 'Maximum number of resources ({0}) exceeded ({1})'
```
Comment (kddejong) [inline src/cfnlint/rules/resources/LimitNumber.py]: Does the language seem backwards with the parameters?

Maximum number of resources (# of Resources in the template) exceeded (the Resource LIMIT)?
or how about
The number of resources (len(resources)) exceeded the limit of (resource limit)?

**Comment by fatbasstard** [comment]
Good one, I'll clarify

**Comment by cmmeyer** [comment]
We should probably do this for all limit rules (parameters, mappings, resources, and outputs).  

@fatbasstard  -- do you want to add them to this commit, or shall I merge this one then update the others?

**Comment by fatbasstard** [comment]
I'll add then, on it

**Comment by cmmeyer** [comment]
#78 should give you all the files that need changing.

**Pull request merged**